mllib/src/test/scala/org/apache/spark/ml/feature/RFormulaSuite.scala

```diff
@@ -32,10 +31,20 @@ class RFormulaSuite extends MLTest with DefaultReadWriteTest {
   def testRFormulaTransform[A: Encoder](
       dataframe: DataFrame,
       formulaModel: RFormulaModel,
-      expected: DataFrame): Unit = {
+      expected: DataFrame,
+      expectedAttributes: AttributeGroup*): Unit = {
+    val resultSchema = formulaModel.transformSchema(dataframe.schema)
+    assert(resultSchema.json == expected.schema.json)
```

Comment:
You compare `schema.json` instead of `schema.toString`. Are you sure they have the same effect ?

Reply:
I know they are different but 'schema.json' based compare is more restrictive: it contains the metadata as well.